### ISSUE TITLE
Support for coalesce function

### ISSUE THREAD

**Issue by anuchak** [kuzudb/kuzu#1195]
https://neo4j.com/docs/cypher-manual/current/functions/scalar/#functions-coalesce

Currently we don't support this function, it is present heavily in the SNB query workloads.

**Comment by acquamarin** [comment]
This is how duckdb implements `coalesce`:
Instead of implementing the actual coalesce,  they transformed `coalesce` to `case` expression:
```
// COALESCE(a,b,c) returns the first argument that is NOT NULL, so
// rewrite into CASE(a IS NOT NULL, a, CASE(b IS NOT NULL, b, c))
```

**Comment by andyfengHKU** [comment]
Fixed in #3235
